packages/coordinator/src/services/communication/mtls-spine-client.ts

```diff
@@ -0,0 +1,197 @@
+import {spine} from "@models"
+import axios, {AxiosError, AxiosResponse, RawAxiosRequestHeaders} from "axios"
+import pino from "pino"
+import {serviceHealthCheck, StatusCheckResponse} from "../../utils/status"
+import {addEbXmlWrapper} from "./ebxml-request-builder"
+import {SpineClient} from "./spine-client"
+import {Agent} from "https"
+
+const SPINE_URL_SCHEME = "https"
+const SPINE_ENDPOINT = process.env.TARGET_SPINE_SERVER
+const SPINE_PATH = "Prescription"
+const BASE_PATH = process.env.BASE_PATH
+
+const logger = pino()
+logger.info(`SPINE_ENDPOINT set to: ${SPINE_ENDPOINT}`)
+
+const getClientRequestHeaders = (interactionId: string, messageId: string) => {
+  return {
+    "Content-Type": "multipart/related;" +
+      " boundary=\"--=_MIME-Boundary\";" +
+      " type=text/xml;" +
+      " start=ebXMLHeader@spine.nhs.uk",
+    "SOAPAction": `urn:nhs:names:services:mm/${interactionId}`,
+    "NHSD-Request-ID": messageId
+  }
+}
+
+export class MtlsSpineClient implements SpineClient {
+  private readonly spineEndpoint: string
+  private readonly spinePath: string
+  private readonly ebXMLBuilder: (spineRequest: spine.SpineRequest) => string
+  private httpsAgent: Agent
+
+  constructor(
+    spineEndpoint: string = null,
```

Comment:
I think this is more idiomatic:
```typescript
    spineEndpoint?: string,
    spinePath?: string,
    ebXMLBuilder?: (spineRequest: spine.SpineRequest) => string
```
since you immediately check for values

Reply:
We will keep it similar to `live-spine-client.ts`